core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -437,6 +487,67 @@ public PrivateKey getDefaultPrivateKey() throws Exception {
                 "available only for the super tenant.");
     }
 
+    /**
+     * This method loads the private key of a given tenant key store.
+     *
+     * @param keyStoreName  Name of the tenant key store.
+     * @param alias         Alias of the private key.
+     * @return Private key corresponding to the alias.
+     */
+    private PrivateKey getTenantPrivateKey(String keyStoreName, String alias) throws Exception {
```

Comment:
Better to use specific Exceptions rather than using "Exception"

Reply:
This method is extracted from the previous `getKeyStore` method.
The logic within throws `RegistryException`, `KeyStoreException`, `CryptoException`, `IOException`, and `SecurityException`.

Although we can catch these and throw a single type of exception, it could disrupt other flows where these errors were handled. Therefore opted out from changing this